protocol/xmrtaker/swap_state.go

```diff
@@ -452,29 +451,21 @@ func (s *swapState) lockETH(amount common.EtherAmount) (ethcommon.Hash, error) {
 	if s.ethAsset != types.EthAssetETH {
 		// TODO: check logs
 		// TODO: check units
-
-		// check that the approval is required
 		// TODO: separate to its own function and create unit tests
 		token, err := contracts.NewIERC20(s.ethAsset.Address(), s.EthClient())
 		if err != nil {
-			log.Errorf("failed to instantiate IERC20: %s", s.ethAsset)
-			return ethcommon.Hash{}, err
+			return ethcommon.Hash{}, fmt.Errorf("failed to instantiate IERC20: %s", err)
```

Comment:
Do you want to use %w above instead of %s?

Reply:
yep, updated!